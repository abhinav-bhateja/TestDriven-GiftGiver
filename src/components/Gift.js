import React, {Component} from "react";
import {Form, Button} from 'react-bootstrap'

class Gift extends Component {
    constructor() {
        super();

        this.state = {person: '', present: ''}
    }

    render() {
        return (
            <div style={{margin: "2rem"}}>
                <Form>
                    <Form.Group>
                        <Form.Label>Person</Form.Label>
                        <Form.Control className="input-person"
                                      onChange={event => this.setState({person: event.target.value})}/>
                    </Form.Group>
                    <Form.Group>
                        <Form.Label>Present</Form.Label>
                        <Form.Control className="input-present"
                                      onChange={event => this.setState({present: event.target.value})}/>
                    </Form.Group>
                </Form>
                <Button variant="danger" className="btn-remove" onClick={() => {
                    this.props.removeGift(this.props.gift.id)
                }}>Remove Gift</Button>
            </div>
        )
    }
}

export default Gift;