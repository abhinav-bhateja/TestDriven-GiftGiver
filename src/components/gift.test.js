import React from "react";
import {shallow} from "enzyme";
import Gift from './Gift';

describe('Gift', () => {
    const mockRemove = jest.fn();
    const id = 1;
    const props = {gift: {id}, removeGift: mockRemove}
    const gift = shallow(<Gift {...props}/>)

    it('renders correctly', () => {
        expect(gift).toMatchSnapshot();
    });

    it('should initialize a person and present in `state`', function () {
        expect(gift.state()).toEqual({person: '', present: ''});
    });

    describe('when typing into the person input', () => {
        const person = 'Uncle';
        beforeEach(() => {
            gift.find('.input-person').simulate('change', {target: {value: person}});
        });

        it('should update the person in the state', function () {
            expect(gift.state().person).toEqual(person);
        });
    });

    describe('when typing into the present input', () => {
        const present = 'PS5';
        beforeEach(() => {
            gift.find('.input-present').simulate('change', {target: {value: present}});
        });

        it('should update the present in the state', () => {
            expect(gift.state().present).toEqual(present);
        });
    })

    describe('when clicking `remove gift` button', () => {
        beforeEach(() => {
            gift.find('.btn-remove').simulate('click');
        });

        it('should call the remove gift callback', () => {
            expect(mockRemove).toHaveBeenCalledWith(id);
        });
    })

});
