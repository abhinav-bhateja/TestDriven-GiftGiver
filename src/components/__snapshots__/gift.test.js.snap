// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "2rem",
    }
  }
>
  <Form
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Person
      </FormLabel>
      <FormControl
        className="input-person"
        onChange={[Function]}
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Present
      </FormLabel>
      <FormControl
        className="input-present"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
  <Button
    active={false}
    className="btn-remove"
    disabled={false}
    onClick={[Function]}
    variant="danger"
  >
    Remove Gift
  </Button>
</div>
`;
